### PR TITLE
docs: AncestorBindings- fixing example code

### LINKED ISSUE #1206: Bugs in AncestorBinding Docs
https://platform.uno/docs/articles/external/uno.toolkit.ui/doc/helpers/ancestor-itemscontrol-binding.html

We currently don't support AncestorBinding/ItemsControlBinding on non-`FrameworkElement`s like `Run`  (#1204), so the documentation should be suggesting using the AncestorBinding on TextBlocks instead of Runs.

Also, the AncestorBinding in the second example that binds to the Tag of the Border is not working. Because we are hitting the Border inside of the ListView template first. (#1205)

We should perform the Tag lookup on a more unique element, like the Page element instead.

The example XAML should look something like:

```xml
<Page x:Class="AncestorBindingSample.MainPage"
	  xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
	  xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
	  xmlns:local="using:AncestorBindingSample"
	  xmlns:utu="using:Uno.Toolkit.UI"
	  Background="{ThemeResource ApplicationPageBackgroundThemeBrush}"
	  Tag="From Page">
	<Grid RowSpacing="20"
		  HorizontalAlignment="Center">
		<Grid.RowDefinitions>
			<RowDefinition Height="Auto" />
			<RowDefinition Height="*" />
		</Grid.RowDefinitions>

		<TextBlock Text="From DataContext: "
				   FontSize="30"
				   FontWeight="Bold">
			<TextBlock.Inlines>
				<Run Text="{Binding SomeText}" />
			</TextBlock.Inlines>
		</TextBlock>
		<ListView ItemsSource="{Binding Items}"
				  Grid.Row="1"
				  Tag="From ListView">
			<ListView.ItemTemplate>
				<DataTemplate>
					<StackPanel>
						<TextBlock Text="Current item DataContext:" />
						<TextBlock FontWeight="Bold"
								   Text="{Binding}" />
						<TextBlock Text="Current parent (MainViewModel) DataContext:" />
						<TextBlock FontWeight="Bold"
								   Text="{utu:ItemsControlBinding Path=DataContext.SomeText}" />
						<TextBlock Text="Accessing property of a parent ListView:" />
						<TextBlock FontWeight="Bold"
								   Text="{utu:ItemsControlBinding Path=Tag}" />
						<TextBlock Text="Accessing property of a Page:" />
						<TextBlock FontWeight="Bold"
								   Text="{utu:AncestorBinding AncestorType=Page, Path=Tag}" />
					</StackPanel>
				</DataTemplate>
			</ListView.ItemTemplate>
		</ListView>
	</Grid>
</Page>
```

We should validate that this example XAML works on all supported platforms

### PR DESCRIPTION
**Pull request by Kunal22shah**
GitHub Issue (If applicable): closes #1206

## PR Type

What kind of change does this PR introduce?
- Documentation content changes

## PR Checklist

Please check if your PR fulfills the following requirements:
- [ ] Tested code with current [supported SDKs](../README.md#supported)
- [ ] Tested the changes where applicable:
	- [ ] UWP
	- [ ] WinUI
	- [ ] iOS
	- [ ] Android
	- [ ] WASM
	- [ ] MacOS
- [ ] Updated the documentation as needed:
	- [ ] [General Doc Update](https://github.com/unoplatform/uno.toolkit.ui/tree/main/doc)
	- [ ] [Controls Doc Update](https://github.com/unoplatform/uno.toolkit.ui/tree/main/doc/controls)
	- [ ] [Extensions Doc Update](https://github.com/unoplatform/uno.toolkit.ui/tree/main/doc/helpers)
	- [ ] [controls-styles.md](https://github.com/unoplatform/uno.toolkit.ui/blob/main/doc/controls-styles.md)
	- [ ] [lightweight-styling.md (LightWeight Styling Resource Keys)](https://github.com/unoplatform/uno.toolkit.ui/blob/main/doc/lightweight-styling.md)
- [ ] [Runtime Tests and/or UI Tests](https://platform.uno/docs/articles/contributing/guidelines/creating-tests.html) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal)
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Review by Xiaoy312 — APPROVED** [review]

File: doc/helpers/ancestor-itemscontrol-binding.md [hunk 1]
```diff
@@ -81,25 +81,49 @@ xmlns:utu="using:Uno.Toolkit.UI"
...
+		<ListView ItemsSource="{Binding Items}"
+				  Grid.Row="1"
+				  Tag="From ListView">
+			<ListView.ItemTemplate>
+				<DataTemplate>
+					<StackPanel>
+						<TextBlock Text="Current item DataContext:" />
+						<TextBlock FontWeight="Bold"
+								   Text="{Binding}" />
+						<TextBlock Text="Current parent (MainViewModel) DataContext:" />
```
Comment (kazo0) [inline doc/helpers/ancestor-itemscontrol-binding.md:114]:
```suggestion
						<TextBlock Text="Current parent (ExampleModel) DataContext:" />
```

**Review by kazo0 — APPROVED** [review]

File: doc/helpers/ancestor-itemscontrol-binding.md
Comment (kazo0) [inline doc/helpers/ancestor-itemscontrol-binding.md:1]: Looks like the markdown validation is failing, you will need to convert the tabs to spaces in the code block:
image

File: doc/helpers/ancestor-itemscontrol-binding.md [hunk 2]
```diff
@@ -81,25 +81,45 @@ xmlns:utu="using:Uno.Toolkit.UI"
...
+                        <TextBlock Text="Accessing property of a parent ListView:" />
+                        <TextBlock FontWeight="Bold" Text="{utu:ItemsControlBinding Path=Tag}" />
+                        <TextBlock Text="Accessing property of a Page:" />
+                        <TextBlock FontWeight="Bold" Text="{utu:AncestorBinding AncestorType=Page, Path=Tag}" />
+                    </StackPanel>
+                </DataTemplate>
+            </ListView.ItemTemplate>
+        </ListView>
+    </Grid>
+</Page>
```
Comment (Xiaoy312) [inline doc/helpers/ancestor-itemscontrol-binding.md:124]: avoid dumping the full xaml here
just keep it short and straight to the point:
```suggestion
<Page Tag="From Page">
    <StackPanel>
        <TextBlock Text="From DataContext:" />
        <TextBlock Text="{Binding SomeText}" />

        <ListView ItemsSource="{Binding Items}" Tag="From ListView">
            <ListView.ItemTemplate>
                <DataTemplate>
                    <StackPanel>
                        <TextBlock Text="Current item DataContext:" />
                        <TextBlock FontWeight="Bold" Text="{Binding}" />

                        <TextBlock Text="Current parent (ExampleModel) DataContext:" />
                        <TextBlock FontWeight="Bold" Text="{utu:ItemsControlBinding Path=DataContext.SomeText}" />

                        <TextBlock Text="Accessing property of a parent ListView:" />
                        <TextBlock FontWeight="Bold" Text="{utu:ItemsControlBinding Path=Tag}" />

                        <TextBlock Text="Accessing property of a Page:" />
                        <TextBlock FontWeight="Bold" Text="{utu:AncestorBinding AncestorType=Page, Path=Tag}" />
                    </StackPanel>
                </DataTemplate>
            </ListView.ItemTemplate>
        </ListView>
    </Grid>
</Page>
```
we can forget the xmlnses... it was mentioned right above

**Comment by agneszitte** [comment]
@Mergifyio backport release/stable/6.1

**Pull request merged**